var/spack/repos/builtin/packages/py-pyfr/package.py

```diff
@@ -24,46 +24,55 @@ class PyPyfr(PythonPackage, CudaPackage, ROCmPackage):
     version("master", branch="master")
 
     # pypi releases
+    version(
+        "2.0.2",
+        sha256="2c6bf460ffec446a933451792c09d3cd85d6703f14636d99810d61823b8d92c7",
+        preferred=True,
+    )
     version("1.15.0", sha256="6a634b9d32447f45d3c24c9de0ed620a0a0a781be7cc5e57b1c1bf44a4650d8d")
     version("1.14.0", sha256="ebf40ce0896cce9ac802e03fd9430b5be30ea837c31224531a6d5fd68f820766")
     version("1.13.0", sha256="ac6ecec738d4e23799ab8c50dea9bdbd7d37bc971bd33f22720c5a230b8e7b2f")
 
-    variant("metis", default=True, description="Metis for mesh partitioning")
-    variant("scotch", default=False, description="Scotch for mesh partitioning")
+    variant("metis", default=False, when="@:1.15.0", description="Metis for mesh partitioning")
+    variant("scotch", default=True, description="Scotch for mesh partitioning")
     variant("cuda", default=False, description="CUDA backend support")
     variant("hip", default=False, description="HIP backend support")
     variant("libxsmm", default=True, description="LIBXSMM for OpenMP backend")
-    variant("scipy", default=True, description="Scipy acceleration for point sampling")
```

Comment:
Should this be a conditional dependency, say for when "@1.14.0:1.15.0"?

Reply:
As this was always used greedily only in relatively rare circumstances, and in a version that we do not expect users to use I think it is not needed.